x-pack/platform/plugins/shared/observability_ai_assistant/common/convert_messages_for_inference.ts

```diff
@@ -28,22 +29,27 @@ function safeJsonParse(jsonString: string | undefined, logger: Pick<Logger, 'err
 
 export function convertMessagesForInference(
   messages: Message[],
-  logger: Pick<Logger, 'error'>
+  logger: Pick<Logger, 'error'>,
+  tools: Record<string, ToolDefinition> | undefined
 ): InferenceMessage[] {
   const inferenceMessages: InferenceMessage[] = [];
 
   messages.forEach((message) => {
     if (message.message.role === MessageRole.Assistant) {
+      const functionCall = message.message.function_call;
+      const isToolAvailable = !!(functionCall?.name && tools?.[functionCall.name]);
       inferenceMessages.push({
         role: InferenceMessageRole.Assistant,
-        content: message.message.content ?? null,
-        ...(message.message.function_call?.name
+        content:
+          message.message.content ??
+          '' + (!isToolAvailable ? ` the tool ${functionCall?.name} is not available` : ''),
+        ...(functionCall?.name
```

Comment:
Can you put this in the tool response, not the content accompanying the tool call?